src/main/java/duke/task/TaskList.java

```diff
@@ -0,0 +1,102 @@
+package duke.task;
+
+import duke.exceptions.IllegalParamException;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Iterator;
+
+/**
+ * Represents a collection of tasks and contains methods to manipulate this collection.
+ */
+public class TaskList implements Iterable<Task>{
+    /** Stores tasks in a list */
+    private List<Task> taskList = new ArrayList<>();
```

Comment:
listOfTasks or tasksList (plural names) to indicate that it is a collection of task might be clearer.